File: java/Clothing.java
class Clothing extends Product 
{
    protected String size;
    protected String material;
    protected String gender;

    public Clothing(int idProduct, String name, String brand, String price, String size, String material, String gender)
    {
        super(idProduct, name, brand, price);
        this.size = size;
        this.material = material;
        this.gender = gender;
    }

    public String toTableRow()
    {
        return String.format("| %-7d | %-15s | %-10s | %-12s | %-4s | %-12s | %-6s |", idProduct, name, brand, price, size, material, gender);
    }
}

File: py/clothing.py
from product import Product

class Clothing(Product):
    def __init__(self, idProduct, name, brand, price, size, material, gender):
        super().__init__(idProduct, name, brand, price)
        self.size = size
        self.material = material
        self.gender = gender

    def toTableRow(self):
        return f"| {self.idProduct:<2} | {self.name:<12} | {self.brand:<8} | {self.price:<10} | {self.size:<4} | {self.material:<10} | {self.gender:<6} |"